tests/x11/oomath.pm

```diff
@@ -33,18 +33,20 @@ sub run {
     my $retries = 7;
     for (1 .. $retries) {
         type_string_slow "E %PHI = H %PHI\nnewline\n1 = 1";
+        wait_still_screen(2);
         last                                                             if check_screen [qw(test-oomath-1 oomath-bsc1127895)];
         die "Could not match on correct formula within multiple retries" if $_ == $retries;
         record_info 'workaround', 'retrying unstable formula typing, see https://progress.opensuse.org/issues/53795 for details';
         send_key 'ctrl-a';
         send_key 'delete';
     }
-    if (match_has_tag('oomath-bsc1127895')) {
-        record_soft_failure 'bsc#1127895';
-        send_key 'alt-f4';
-    } else {
-        send_key 'alt-f4';
-        assert_and_click 'dont-save-libreoffice-btn';    # _Don't save
+
+    send_key 'alt-f4';
+    if (check_screen('dont-save-libreoffice-btn')) {
+        assert_and_click 'dont-save-libreoffice-btn';
+    }
+    elsif (check_screen('generic-desktop')) {
+        record_soft_failure('bsc#1184961');
     }
```

Comment:
Can you use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen`?

Reply:
changed